### PR TITLE
Hotfix/search accumulation revamp

### PR DESCRIPTION
**Pull request by Ravikant96**
Description:
Describe your pull request and attach links to any issues that may be related.

### CODE REVIEW

File: client/js/reports.js [hunk 1]
```diff
@@ -1350,30 +1350,75 @@ class DataSourceColumns extends Map {
...
-			'#6c54b5',
-			'#bb7784',
-			'#b5bbe3',
-			'#0c8765',
-			'#ef9708',
-			'#1abb9c',
-			'#9da19c',
-		];
+		this.column = column;
+		this.filters = this.column.filters && this.column.filters.length ? this.column.filters : [{name:'0', value:''}];
```
Comment (achshar) [inline client/js/reports.js]: 1. Spacing
2. `const`

File: client/js/reports.js [hunk 2]
```diff
@@ -1350,30 +1350,75 @@ class DataSourceColumns extends Map {
 	}
 }
 
-class DataSourceColumn {
+class DataSourceColumnFilters {
```
Comment (achshar) [inline client/js/reports.js]: Extends `Set`

File: client/js/reports.js [hunk 3]
```diff
@@ -1350,30 +1350,75 @@ class DataSourceColumns extends Map {
...
+		const container = this.containerElement = document.createElement('div');
+
+		container.classList.add('show', 'filters');
+
+		container.innerHTML = `
+			<span>
+				Search
+				<button type="button" class="show add-filter add-new-item"><i class="fa fa-plus"></i></button>
+			</span>
+			<div class="filter-container"></div>
```
Comment (achshar) [inline client/js/reports.js]: `class="list"`

File: client/js/reports.js [hunk 4]
```diff
@@ -1350,30 +1350,75 @@ class DataSourceColumns extends Map {
...
+		`;
+
+		container.querySelector('button.add-filter').on('click', () => {
+
+			this.list.add(new DataSourceColumnFilter({name:'0', value:''}, this));
```
Comment (achshar) [inline client/js/reports.js]: Spacing

File: client/js/reports.js [hunk 5]
```diff
@@ -1350,30 +1350,75 @@ class DataSourceColumns extends Map {
...
+		return container;
+	}
+
+	render() {
+
+		const div = this.container.querySelector('.filter-container');
+
+		div.textContent = null;
+
+		for(const filter of this.list)
```
Comment (achshar) [inline client/js/reports.js]: Show "No filters added" if no filters are added.

File: client/js/reports.js [hunk 6]
```diff
@@ -1350,30 +1350,75 @@ class DataSourceColumns extends Map {
...
+			div.appendChild(filter.container);
+	}
+
+	get json() {
+
+		const json = [];
+
+		for(const filter of this.list) {
+			if(filter.json.value != '')
+				json.push(filter.json)
```
Comment (achshar) [inline client/js/reports.js]: semi colon

File: client/js/reports.js [hunk 7]
```diff
@@ -1419,8 +1464,110 @@ class DataSourceColumn {
 				apply: (q, v) => q.toString().match(new RegExp(q, 'i')),
 			},
 		];
+	}
+
+	constructor(filter, filters) {
+
+		for(const key in filter)
```
Comment (achshar) [inline client/js/reports.js]: `Object.assign`

File: client/js/reports.js [hunk 8]
```diff
@@ -1419,8 +1464,110 @@ class DataSourceColumn {
...
+			this.filters.list.delete(this);
+			this.filters.render();
+		});
+
+		return container;
+	}
+
+	get json() {
+
+		return {name: this.container.querySelector('select').value, value: this.container.querySelector('input').value};
```
Comment (achshar) [inline client/js/reports.js]: Expand

File: client/js/reports.js [hunk 9]
```diff
@@ -1419,8 +1464,110 @@ class DataSourceColumn {
...
+	}
+}
+
+class DataSourceColumnAccumulations {
+
+	constructor(column) {
+
+		this.column = column;
+		this.accumulations =[{name:'', value:''}];
```
Comment (achshar) [inline client/js/reports.js]: Space

File: client/js/reports.js [hunk 10]
```diff
@@ -1419,8 +1464,110 @@ class DataSourceColumn {
...
+		return container;
+	}
+
+	get json() {
+
+		return {name: this.container.querySelector('select').value, value: this.container.querySelector('input').value};
+	}
+}
+
+class DataSourceColumnAccumulations {
```
Comment (achshar) [inline client/js/reports.js]: Extends `Set`

File: client/js/reports.js [hunk 11]
```diff
@@ -1419,8 +1464,110 @@ class DataSourceColumn {
...
+		const container = this.containerElement = document.createElement('div');
+
+		container.classList.add('show', 'accumulations');
+
+		container.innerHTML = `
+			<span>
+				Accumulation
+				<button type="button" class="show add-accumulation add-new-item"><i class="fa fa-plus"></i></button>
+			</span>
+			<div class="accumulation-container"></div>
```
Comment (achshar) [inline client/js/reports.js]: `class="list"`

File: client/js/reports.js [hunk 12]
```diff
@@ -1448,6 +1595,108 @@ class DataSourceColumn {
 				string: true,
 			},
 		];
+	}
+
+	constructor(accumulation, accumulations) {
+
+		for(const key in accumulation)
```
Comment (achshar) [inline client/js/reports.js]: `Object.assign`

File: client/js/reports.js [hunk 13]
```diff
@@ -1448,6 +1595,108 @@ class DataSourceColumn {
...
+		const accumulation = DataSourceColumnAccumulation.accumulationTypes[select.value];
+
+		if(accumulation)
+			this.container.querySelector('input').value = accumulation.apply(this.accumulations.column.source.response, this.accumulations.column.key);
+
+		else this.container.querySelector('input').value = '';
+	}
+}
+
+class DataSourceColumn {
```
Comment (achshar) [inline client/js/reports.js]: Move class up

File: client/js/reports.js [hunk 14]
```diff
@@ -1719,22 +1963,8 @@ class DataSourceColumn {
...
-			form.querySelector('.filters').appendChild(this.searchBox());
-		});
-
-		form.querySelector('.accumulations').appendChild(this.accumulationBox());
-
-		form.querySelector('.add-accumulation').on('click', e => {
-
-			form.querySelector('.accumulations').appendChild(this.accumulationBox());
-		});
+		form.insertBefore(this.columnFilters.container,form.querySelector('.columnType'));
```
Comment (achshar) [inline client/js/reports.js]: spacing

File: client/js/reports.js (hunk 13 above)
Comment (Ravikant96) [reply client/js/reports.js]: Its done. But not reflecting here. May be a github bug.

**Pull request merged**